### PR TITLE
Remove weird Rational dispatch and add pi functions to list

### LINKED ISSUE #48735: `sinpi(::Rational{BigInt})` regression
#41744 broke `sinpi(::Rational{BigInt})`, which worked in v1.6, v1.8 and v1.9-beta:

Before:
```
julia> sinpi(big(1//1))
0.0
```
After #41744:
```
julia> sinpi(big(1//1))
ERROR: MethodError: no method matching sinpi_kernel(::BigFloat)
[...]
```

### PR DESCRIPTION
**Pull request by gbaraldi**
Should fix  https://github.com/JuliaLang/julia/issues/48735

### CODE REVIEW

**Comment by oscardssmith** [comment]
This is mostly good but I have 2 nits. The first is that there's now some useless complexity in the `sinpi(_x::IEEEFloat)` methods from when they also needed to support `Rational`. The second is that it probably would be a good idea to change the `IEEEFloat` methods to `AbstractFloat` methods and define `sinpi_kernel(x::AbstractFloat) = sin_kernel(x *pi)` etc. This would increase accuracy since the multiplication by `pi` would be done after the reduction. (If you do this though, keep in mind that `maxintfloat(::BigFloat)` isn't a fixed number).

**Comment by gbaraldi** [comment]
BigFloat should never take this path anyway, mpfr has `sinpi` and friends already.

File: test.jl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+using Random: seed!
```
Comment (oscardssmith) [inline test.jl:2]: what's this?

Comment (gbaraldi) [reply test.jl:2]: Garbage!

**Comment by sostock** [comment]
This does resolve #48735, but the fact that the conversion from rational to float now happens before the range reduction introduces new inaccuracies:

Julia 1.9:
```julia
julia> sinpi(typemax(Int)//2)
-1.0
```

This PR:
```julia
julia> sinpi(typemax(Int)//2) # conversion to Float64 results in an integer value
0.0
```

**Comment by gbaraldi** [comment]
That's annoying, because the float cast makes us avoid the overflow that happens with
```julia
sinpi(typemax(Int64)//1)
```
But it makes us lose precision because we don't do range reduction, except that it is the range reduction that overflows 😮‍💨

**Comment by gbaraldi** [comment]
1.9 was already broken in a similar case to this, if your number isn't really representable with a float64 and you want to get an accurate float64 at the end I'm not sure if there is anything we can do except recommending that you switch to BigNums

**Comment by oscardssmith** [comment]
imo this is ready to merge

**Pull request merged**